tests_e2e/tests/ext_policy/ext_policy.py

```diff
@@ -0,0 +1,225 @@
+        """
+        with open("waagent_policy.json", mode='w') as policy_file:
+            json.dump(policy, policy_file, indent=4)
+            policy_file.flush()
+
+            remote_path = "/tmp/waagent_policy.json"
+            local_path = policy_file.name
+            self._ssh_client.copy_to_node(local_path=local_path, remote_path=remote_path)
+            policy_file_final_dest = "/etc/waagent_policy.json"
+            log.info("Copying policy file to test VM [%s]", self._context.vm.name)
+            self._ssh_client.run_command(f"mv {remote_path} {policy_file_final_dest}", use_sudo=True)
+
+    def _operation_should_succeed(self, operation, extension_case):
+        log.info(f"Attempting to {operation} {extension_case.extension.__str__()}, expected to succeed ")
+        # Attempt operation. If enabling, assert that the extension is present in instance view.
+        # If deleting, assert that the extension is not present in instance view.
+        try:
+            if operation == "enable":
+                extension_case.extension.enable(settings=extension_case.settings, force_update=True, timeout=15 * 60)
+                extension_case.extension.assert_instance_view()
+            elif operation == "delete":
+                extension_case.extension.delete(timeout=15 * 60)
+                instance_view_extensions = self._context.vm.get_instance_view().extensions
+                if instance_view_extensions is not None and any(
+                        e.name == extension_case.extension._resource_name for e in instance_view_extensions):
+                    raise Exception(
+                        "extension {0} still in instance view after attempting to delete".format(extension_case.extension._resource_nam))
+            log.info(f"Operation '{operation}' for {extension_case.extension.__str__()} succeeded as expected.")
+        except Exception as error:
+            fail(
+                f"Unexpected error while trying to {operation} {extension_case.extension.__str__()}. "
+                f"Extension is allowed by policy so this operation should have completed successfully.\n"
+                f"Error: {error}")
+
+    @staticmethod
+    def _operation_should_fail(operation, extension_case):
+        log.info(f"Attempting to {operation} {extension_case.extension.__str__()}, should fail fast.")
+        try:
+            timeout = (6 * 60)  # Fail fast.
+            if operation == "enable":
+                extension_case.extension.enable(settings=extension_case.settings, force_update=True, timeout=timeout)
+            elif operation == "delete":
+                extension_case.extension.delete(timeout=timeout)
+            fail(f"The agent should have reported an error trying to {operation} {extension_case.extension.__str__()} "
+                 f"because the extension is disallowed by policy.")
+        except Exception as error:
+            assert_that("Extension will not be processed" in str(error)) \
```

Comment:
Could we also check for [ExtensionPolicyError] in the message to confirm the failure was due to policy

Reply:
updated